ProjectAnswers.tex

```diff
@@ -462,7 +506,64 @@ \subsubsection{Reduction to Upper Hessenberg Form}
             Thus, the upper Hessenberg form of a symmetric matrix naturally becomes tridiagonal because the symmetry constraint forces zeros to appear below and above the second subdiagonals.
         \end{proof}
 \end{enumerate}
-\subsection{2.1 R}
+\subsection{Solution: Proof that \( A_{k-1} \) and \( A_k \) have the same eigenvalues}
+\textbf{Objective:}  
+We aim to prove that for every \( k \geq 1 \), the matrices \( A_{k-1} \) and \( A_k \) generated by the "Practical" QR Algorithm have the same eigenvalues.
+
+\subsection*{Algorithm Description}  
+The "Practical" QR Algorithm with shifts generates a sequence of matrices \( A_0, A_1, A_2, \dots \) as follows:
+1. Start with the initial matrix \( A_0 \), which is the upper Hessenberg form of the original matrix \( A \).
+2. For each iteration \( k \), a shift \( \mu_k \) is chosen, typically an eigenvalue approximation of \( A_{k-1} \).
+3. The shifted matrix \( A_{k-1} - \mu_k I \) undergoes QR factorization:
+   \[
+   Q_k R_k = A_{k-1} - \mu_k I,
+   \]
+   where \( Q_k \) is orthogonal (\( Q_k^\top Q_k = I \)) and \( R_k \) is upper triangular.
+4. The next matrix \( A_k \) is computed as:
+   \[
+   A_k = R_k Q_k + \mu_k I.
+   \]
+
+We will now show that \( A_{k-1} \) and \( A_k \) have the same eigenvalues for all \( k \).
+
+\subsection*{Step 1: QR Factorization and Eigenvalue Preservation}  
+The matrix \( A_{k-1} - \mu_k I \) is factored as:
+\[
+A_{k-1} - \mu_k I = Q_k R_k,
+\]
+where \( Q_k \) is orthogonal. Multiplying both sides on the left by \( Q_k^\top \), we get:
+\[
+Q_k^\top (A_{k-1} - \mu_k I) = R_k.
+\]
+Since \( Q_k \) is orthogonal, the eigenvalues of \( A_{k-1} - \mu_k I \) are preserved under this transformation. So, \( Q_k R_k \) and \( A_{k-1} - \mu_k I \) have the same eigenvalues.
+
+\subsection*{Step 2: Transformation from \( A_{k-1} \) to \( A_k \)}  
+The next matrix \( A_k \) is defined as:
+\[
+A_k = R_k Q_k + \mu_k I.
+\]
+Substituting \( R_k = Q_k^\top (A_{k-1} - \mu_k I) \), we can rewrite \( A_k \) as:
+\[
+A_k = Q_k^\top (A_{k-1} - \mu_k I) Q_k + \mu_k I.
+\]
+Simplifying this expression:
+\[
+A_k = Q_k^\top A_{k-1} Q_k - Q_k^\top (\mu_k I) Q_k + \mu_k I.
+\]
+Since \( Q_k^\top Q_k = I \), this simplifies further to:
+\[
+A_k = Q_k^\top A_{k-1} Q_k.
+\]
+
+\subsection*{Step 3: Similarity Transformations Preserve Eigenvalues}  
+Two matrices \( A \) and \( B \) are similar if there exists an invertible matrix \( P \) such that:
+\[
+B = P^{-1} A P.
+\]
+It is a fundamental property of matrices that similar matrices have the same eigenvalues. In our case, \( A_k = Q_k^\top A_{k-1} Q_k \) shows that \( A_k \) is similar to \( A_{k-1} \), since \( Q_k \) is orthogonal and hence invertible.
```

Comment:
I think this is much wordier than it needs to be especially since we already discussed similar matricies and how they share eigenvalues in 1.1.
```suggestion
Since $Q$ is orthogonal, $Q^{-1} = Q^T$. Therefore, $A_k $ and $A_{k-1}$ are similar matrices and thus share eigenvalues as proven in \textbf{1.1}.
```